File: app/src/main/java/com/safetyheads/akademiaandroida/utils/EmailValidator.java
package com.safetyheads.akademiaandroida.utils;

import android.util.Patterns;
import android.widget.EditText;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;


public class EmailValidator extends AppCompatActivity {

    public void emailValidator(EditText etMail) {
        String emailToText = etMail.getText().toString();
        if (!emailToText.isEmpty() && Patterns.EMAIL_ADDRESS.matcher(emailToText).matches()) {
            Toast.makeText(this, "Email Verified !", Toast.LENGTH_SHORT).show();
        } else {
            Toast.makeText(this, "Enter valid Email address !", Toast.LENGTH_SHORT).show();
        }
    }

}
//To use this code you have to use in xml file Button (android:id="@+id/validateButton") and EditText (android:id="@+id/emailField")

File: presentation/src/main/java/com/safetyheads/akademiaandroida/presentation/ui/utils/EmailValidator.kt
package com.safetyheads.akademiaandroida.presentation.ui.utils

import android.widget.EditText
import androidx.core.util.PatternsCompat
import androidx.core.widget.addTextChangedListener
import com.safetyheads.akademiaandroida.R

object EmailValidator {
    fun attach(editText: EditText) {
        editText.addTextChangedListener {
            validateEmail(it.toString(), editText)
        }
    }

    fun validateEmail(email: String, editText: EditText) {
        if (!isValid(email)) {
            editText.error = editText.context.getString(R.string.invalid_email)
        } else {
            editText.error = null
        }
    }

    fun isValid(email: String): Boolean {
        return PatternsCompat.EMAIL_ADDRESS.matcher(email).matches()
    }

}